<div class="alert alert-block alert-info">
<center>ESTATÍSTICA APLICADA</center>
<center>Projeto I</center>
<center>ALUNO: Diego Cardoso dos Santos</center>
</div>

### 1ª Tarefa ###

Carregue em dataframes os conjuntos de dados referentes ao ano de 2018, 2019 e 2020. Corrija quaisquer erros que apareçam no carregamento dos arquivos. É possível acessar os arquivos compactados diretamente do site, sem precisar baixá-los?

Escolhi trabalhar os casos "Agrupados por ocorrência", como o foco do trabalho é os acidentes, acreditei que o arquivo .csv já tem as especificações necessárias para os tipos de acidente e para as tarefas do projeto em sua totalidade.

Resposta: Sim, há como abrir os arquivos diretamente do site utilizando o método compression='zip', definição da própria biblioteca: "Se str, representa o modo de compressão. Se dict, o valor em 'método' é o modo de compressão. O modo de compressão pode ser qualquer um dos seguintes valores possíveis: {'inferir', 'gzip', 'bz2', 'zip', 'xz', Nenhum}." Dessa forma, usando o read_csv, com o compression='zip', é possível analisar o arquivo .csv dentro deste.

Após isso, iniciei o processo de filtragem, o primeiro erro apresentado foi no UnicodeDecodeError, pois o padrão 'utf-8', não suporta caracteres especias presentes nas línguas latinas, então mudei o enconding para 'latin1', e então consegui resolver esse problema, após isso, o dataframe saiu de maneira estranha, então descobri que o separador do arquivo .csv era por ponto e vírgula ";", então, alterei este: " sep=';' ", e então os dataframes deram certo.

OBS: Coloquei a primeira forma em comentário que fiz tentando resolver a abertura do arquivo zipado, não consegui na primeira tentativa abrir diretamente pelo site, então fui tentar abrir com a biblioteca zipFile, abri o arquivo .zip primeiramente e depois mandei ele printar todos os arquivos que continham neste, dessa forma, vi que só tinha apenas o "acidentes2017_todas_causas_tipos.csv", então, posteriormente abri esse arquivo .csv e utilizei o read e as filtragens.

In [17]:
import pandas as pd
import numpy as np

In [2]:
"""
import zipfile

with zipfile.ZipFile('C:/Users/diego/para_anaconda/projeto_1/acidentes2017_todas_causas_tipos.zip') as z:
    print(*z.namelist(),sep="\n")
    
with zipfile.ZipFile('C:/Users/diego/para_anaconda/projeto_1/acidentes2017_todas_causas_tipos.zip') as z:
    with z.open('acidentes2017_todas_causas_tipos.csv') as f:
        df = pd.read_csv(f, sep=';', encoding='latin1')

"""

ano_2018 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/MaC6cieXSFACNWT/download', compression='zip', sep=';', encoding='latin1')
ano_2019 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/kRBUylqz6DyQznN/download', compression='zip', sep=';', encoding='latin1')
ano_2020 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/jdDLrQIf33xXSCe/download', compression='zip', sep=';', encoding='latin1')


In [3]:
print()
print("Ano 2018")
print("________________________________________")
display(ano_2018)
print()
print("Ano 2019")
print("________________________________________")
display(ano_2019)
print()
print("Ano 2020")
print("________________________________________")
display(ano_2020)


Ano 2018
________________________________________


id data_inversa     dia_semana   horario  uf     br     km  \
0       99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1       99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
2       99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
3       99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
4       99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...         ...          ...            ...       ...  ..    ...    ...   
69290  252256.0   2018-12-24  segunda-feira  09:15:00  PI  343.0  263,1   
69291  252652.0   2018-12-12   quarta-feira  07:00:00  BA  101.0    321   
69292  252801.0   2018-11-15   quinta-feira  10:48:00  SP  116.0    529   
69293  253040.0   2018-11-23    sexta-feira  19:30:00  RS  472.0  573,5   
69294  253078.0   2018-10-01  segunda-feira  04:31:00  GO   40.0      9   

                       municipio  \
0                        RESENDE   
1                  FLORIANOPOLIS   
2                       ITABORAI   
3                        SARANDI   
4                       CANDIOTA   
...                          ...   
69290                CAMPO MAIOR   
69291  PRESIDENTE TANCREDO NEVES   
69292             BARRA DO TURVO   
69293                 URUGUAIANA   
69294                   LUZIANIA   

                                          causa_acidente  \
0                                      Condutor Dormindo   
1                     Não guardar distância de segurança   
2                                 Ultrapassagem Indevida   
3                                     Ingestão de Álcool   
4                            Falta de Atenção à Condução   
...                                                  ...   
69290                                   Agressão Externa   
69291                        Falta de Atenção à Condução   
69292                        Falta de Atenção à Condução   
69293  Desobediência às normas de trânsito pelo condutor   
69294                 Não guardar distância de segurança   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0         Saída de leito carroçável  ...              4      1         0   
1                  Colisão traseira  ...              2      1         0   
2                   Colisão frontal  ...              0      3         1   
3               Colisão transversal  ...              0      2         0   
4         Saída de leito carroçável  ...              1      0         0   
...                             ...  ...            ...    ...       ...   
69290   Colisão com objeto estático  ...              0      1         0   
69291     Saída de leito carroçável  ...              0      0         0   
69292              Colisão traseira  ...              0      0         1   
69293  Queda de ocupante de veículo  ...              1      1         0   
69294     Saída de leito carroçável  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
1           2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
2           1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
3           0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
4           1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...       ...      ...           ...           ...       ...        ...   
69290       0        1     -4,819326    -42,158821     SR-PI    DEL17/1   
69291       1        1  -13,44768518  -39,42066193     SR-BA    DEL10/5   
69292       1        2    -24,939211    -48,301248     SR-SP     DEL6/5   
69293       1        2  -29,76355779  -57,05200195     SR-RS    DEL9/13   
69294       1        1    -16,131158    -47,957739     SR-DF   UOP01/DF   

            uop  
0      UOP03/RJ  
1      UOP01/SC  
2      UOP02/RJ  
3      UOP01/RS  
4      UOP03/RS  
...         ...  
69290  UOP03/PI  



Ano 2019
________________________________________


id data_inversa     dia_semana   horario  uf     br     km  \
0      182210.0   2019-01-01    terça-feira  01:30:00  SP  116.0    218   
1      182211.0   2019-01-01    terça-feira  01:30:00  PR  373.0  177,3   
2      182212.0   2019-01-01    terça-feira  00:45:00  SC  101.0     16   
3      182214.0   2019-01-01    terça-feira  00:00:00  CE   20.0  429,5   
4      182215.0   2019-01-01    terça-feira  01:00:00  MG  381.0    327   
...         ...          ...            ...       ...  ..    ...    ...   
67441  266255.0   2019-10-07  segunda-feira  09:10:00  ES  101.0  270,5   
67442  266406.0   2019-11-25  segunda-feira  07:20:00  PR  116.0    125   
67443  266434.0   2019-06-03  segunda-feira  19:00:00  CE  222.0    1,4   
67444  266573.0   2019-07-13         sábado  19:35:00  PR  373.0    425   
67445  266627.0   2019-10-22    terça-feira  20:20:00  BA  324.0    608   

                municipio                       causa_acidente  \
0               GUARULHOS          Falta de Atenção à Condução   
1            PONTA GROSSA          Falta de Atenção à Condução   
2                  GARUVA                     Animais na Pista   
3               FORTALEZA  Ingestão de Substâncias Psicoativas   
4                NOVA ERA          Falta de Atenção à Condução   
...                   ...                                  ...   
67441               SERRA          Falta de Atenção à Condução   
67442  FAZENDA RIO GRANDE          Falta de Atenção à Condução   
67443             CAUCAIA   Não guardar distância de segurança   
67444              CANDOI          Defeito Mecânico no Veículo   
67445        SIMOES FILHO                       Defeito na Via   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0       Colisão com objeto estático  ...              0      0         0   
1                  Colisão traseira  ...              0      0         0   
2       Colisão com objeto estático  ...              0      0         0   
3       Colisão com objeto estático  ...              1      0         0   
4               Colisão transversal  ...              1      1         0   
...                             ...  ...            ...    ...       ...   
67441                    Tombamento  ...              1      1         0   
67442               Colisão lateral  ...              0      1         1   
67443              Colisão traseira  ...              1      1         0   
67444     Saída de leito carroçável  ...              1      0         0   
67445  Queda de ocupante de veículo  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1  -23,46052014  -46,48772478     SR-SP     DEL6/1   
1           1        1  -25,05533957  -50,22776753     SR-PR     DEL7/3   
2           1        1      -26,1216      -48,8826     SR-SC     DEL8/3   
3           1        1      -3,76999    -38,670063     SR-CE    DEL16/1   
4           1        2      -19,7609      -43,0306     SR-MG     DEL4/3   
...       ...      ...           ...           ...       ...        ...   
67441       1        2     -20,23107     -40,27387     SR-ES    DEL12/2   
67442       1        2     -25,59505     -49,31631     SR-PR     DEL7/1   
67443       1        2     -3,736507     -38,65337     SR-CE    DEL16/1   
67444       1        1      -25,7118     -52,18374     SR-PR     DEL7/3   
67445       1        1     -12,78156     -38,41656     SR-BA    DEL10/1   

            uop  
0      UOP01/SP  
1      UOP01/PR  
2      UOP01/SC  
3      UOP02/CE  
4      UOP01/MG  
...         ...  
67441  UOP01/ES  
67442  UOP03/PR  
67443  UOP01/CE  
67444  UOP02/PR  
67445  UOP01/BA  

[67446 rows x 30 columns]


Ano 2020
________________________________________


id data_inversa    dia_semana   horario  uf     br     km  \
0      260031.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   
1      260036.0   2020-01-01  quarta-feira  01:00:00  RJ  116.0    178   
2      260037.0   2020-01-01  quarta-feira  01:52:00  SC  101.0  206,9   
3      260038.0   2020-01-01  quarta-feira  01:15:00  RO  364.0    236   
4      260039.0   2020-01-01  quarta-feira  01:30:00  PR  153.0    360   
...         ...          ...           ...       ...  ..    ...    ...   
63543  364615.0   2020-03-20   sexta-feira  09:00:00  RN  304.0  171,8   
63544  364810.0   2020-09-12        sábado  23:50:00  BA  116.0   11,5   
63545  364817.0   2020-10-30   sexta-feira  10:30:00  BA  116.0   14,3   
63546  364835.0   2020-09-20       domingo  23:00:00  BA  116.0      8   
63547  364903.0   2020-10-31        sábado  02:10:00  BA  116.0      3   

                  municipio                causa_acidente  \
0                    GURUPI              Animais na Pista   
1               NOVA IGUACU  Falta de Atenção do Pedestre   
2                  SAO JOSE            Ingestão de Álcool   
3                    CACOAL       Velocidade Incompatível   
4                  REBOUCAS            Ingestão de Álcool   
...                     ...                           ...   
63543               ANGICOS   Falta de Atenção à Condução   
63544  VITORIA DA CONQUISTA            Ingestão de Álcool   
63545  VITORIA DA CONQUISTA   Falta de Atenção à Condução   
63546  VITORIA DA CONQUISTA       Velocidade Incompatível   
63547  VITORIA DA CONQUISTA              Animais na Pista   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0           Atropelamento de Animal  ...              5      0         0   
1         Atropelamento de Pedestre  ...              0      1         0   
2                   Colisão lateral  ...              0      1         0   
3         Saída de leito carroçável  ...              1      0         0   
4                       Capotamento  ...              4      0         0   
...                             ...  ...            ...    ...       ...   
63543              Colisão traseira  ...              1      2         0   
63544               Colisão lateral  ...              0      2         0   
63545               Colisão lateral  ...              0      2         0   
63546  Queda de ocupante de veículo  ...              0      0         0   
63547       Atropelamento de Animal  ...              1      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           5        1  -11,77460203  -49,10744996     SR-TO   UOP01/TO   
1           0        1  -22,75223028   -43,4379103     SR-RJ     DEL5/1   
2           1        2  -27,59193546  -48,61824557     SR-SC     DEL8/1   
3           1        1  -11,44624577  -61,43761218     SR-RO    DEL21/2   
4           6        1  -25,67503796  -50,75089805     SR-PR     DEL7/3   
...       ...      ...           ...           ...       ...        ...   
63543       2        4   -5,68711053    -36,419458     SR-RN      UOP03   
63544       1        2  -14,87308214  -40,87141514     SR-BA      UOP01   
63545       2        2  -14,89597653  -40,86098671     SR-BA      UOP01   
63546       1        1  -14,84732305  -40,88634968     SR-BA      UOP01   
63547       1        1    -14,824848    -40,861311     SR-BA      UOP01   

            uop  
0           NaN  
1      UOP02/RJ  
2      UOP01/SC  
3      UOP02/RO  
4      UOP04/PR  
...         ...  
63543     DEL01  
63544     DEL08  
63545     DEL08  
63546     DEL08  
63547     DEL08  

[63548 rows x 30 columns]

### Tafera 2 ###

Explore os conjuntos de dados, respondendo as questões:

    (a) quantos acidentes ocorreram em cada ano?
    (b) quantas variáveis cada conjunto de dados registra?
    (c) quais as cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais?
    (d) quantos acidentes com feridos graves aconteceram na Paraíba em 2019?

In [4]:
"""LETRA A"""

"""
Como o arquivo .csv só trabalha com a quantidade de acidentes, apenas dei um len para saber a quantidade de linhas
para assim representar o número de acidentes.
"""
print("O números de acidente no ano de 2018 foi: {}\n".format(len(ano_2018)))
print("O números de acidente no ano de 2019 foi: {}\n".format(len(ano_2019)))
print("O números de acidente no ano de 2020 foi: {}\n".format(len(ano_2020)))

O números de acidente no ano de 2018 foi: 69295

O números de acidente no ano de 2019 foi: 67446

O números de acidente no ano de 2020 foi: 63548



In [5]:
"""LETRA B"""

"""
Para saber a quantidade de variáveis, dei um len nas colunas dos dataframes.
"""

print("A quantidade de variáveis no dataframe de 2018 foi: {}\n".format(len(ano_2018.columns)))
print("A quantidade de variáveis no dataframe de 2019 foi: {}\n".format(len(ano_2019.columns)))
print("A quantidade de variáveis no dataframe de 2020 foi: {}\n".format(len(ano_2020.columns)))

A quantidade de variáveis no dataframe de 2018 foi: 30

A quantidade de variáveis no dataframe de 2019 foi: 30

A quantidade de variáveis no dataframe de 2020 foi: 30



In [6]:
"""LETRA C"""

"""
Para saber as cidades com maiores acidentes, utilizei o value_counts para contar as ocorrências dos valores e tranformei este em
um dicionário, e reduzi apenas para 5 linhas já que o value já retorna os valores em ordem decrescente.
"""

count_2018 = ano_2018['municipio'].value_counts().head(5).to_dict()
count_2019 = ano_2019['municipio'].value_counts().head(5).to_dict()
count_2020 = ano_2020['municipio'].value_counts().head(5).to_dict()

print("As cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais no ano de 2018 foram:")
for chave, valor in count_2018.items():
    print("{} com {} casos.".format(chave, valor))
print("_______________________________________________________________________________")
print("As cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais no ano de 2019 foram:")
for chave, valor in count_2019.items():
    print("{} com {} casos.".format(chave, valor))
print("_______________________________________________________________________________")
print("As cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais no ano de 2020 foram:")
for chave, valor in count_2020.items():
    print("{} com {} casos.".format(chave, valor))

As cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais no ano de 2018 foram:
CURITIBA com 1019 casos.
SAO JOSE com 874 casos.
BRASILIA com 867 casos.
GUARULHOS com 755 casos.
PALHOCA com 688 casos.
_______________________________________________________________________________
As cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais no ano de 2019 foram:
CURITIBA com 1094 casos.
BRASILIA com 1090 casos.
SAO JOSE com 819 casos.
GUARULHOS com 717 casos.
SERRA com 633 casos.
_______________________________________________________________________________
As cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais no ano de 2020 foram:
BRASILIA com 1040 casos.
CURITIBA com 950 casos.
GUARULHOS com 639 casos.
SAO JOSE com 629 casos.
PALHOCA com 593 casos.


In [7]:
"""LETRA D"""

"""
Nessa questão eu filtrei primeiramente o dataFrame para apenas as colunas dos estados e dos feridos graves, depois disso
filtro novamente para os estados que foram na Paraíba, e por último, fiz retornar o dataframe cujo os valores de feridos graves
fossem maior que zero, deixando apenas os acidentes que tiveram feridos graves na paraíba, então só dei um len para saber
quantos foram precisamente.
"""

df_novo = ano_2019[['uf', 'feridos_graves']]
df_filtrado = df_novo[df_novo['uf'] == 'PB']
df_filtrado = df_filtrado[df_filtrado['feridos_graves'] > 0]
feridos_graves = len(df_filtrado)

print("O número de acidentes com feridos graves foram: {}". format(feridos_graves))

O número de acidentes com feridos graves foram: 414


### Tarefa 3 ###

Para cada indicação abaixo, construa um novo dataframe, salvando-o com a terminação .csv.

    (a) ranking de acidentes por estado para todos os anos (um só arquivo combinando a informação dos três anos);

    (b) acidentes por dia da semana para todos os anos (um só arquivo combinando a informação dos três anos).


In [8]:
"""LETRA A"""

"""
Para essa questão, fiz primeiro 4 dataframes, 3 destes para cada anos respectivos e no último para todos os anos,
utilizei o value_counts nos estados e especifiquei as chaves com .keys() para que mostrasse os primeiros estados
com mais acidentes, assim, com o index, defini os números para a classificação. Por fim, concatenei todos os dataframes
e especifiquei o nome das colunas respectivamente.
Na transformação para csv utilizei o to_csv, mas quando rodava o arquivo com pd.read_csv() o dataframe
saia com Unamed, visto isso, tratei especificando os "names" além de colocar o header=0
ler o arquivo csv que eu criei
"""

todos_anos = pd.concat([ano_2018, ano_2019, ano_2020], ignore_index=True)

ano_2018_filtrado = pd.DataFrame(ano_2018['uf'].value_counts().head(10).keys(), index = list(range(1,11)))
ano_2019_filtrado = pd.DataFrame(ano_2019['uf'].value_counts().head(10).keys(),  index = list(range(1,11)))
ano_2020_filtrado = pd.DataFrame(ano_2020['uf'].value_counts().head(10).keys(),  index = list(range(1,11)))
anos_filtrados = pd.DataFrame(todos_anos['uf'].value_counts().head(10).keys(),  index = list(range(1,11)))

df_concatenado = pd.concat([ano_2018_filtrado, ano_2019_filtrado, ano_2020_filtrado, anos_filtrados],axis = 1)
df_concatenado.columns = ['2018', '2019', '2020', 'Total']

df_concatenado.to_csv('ranking_por_estado.csv')

In [9]:
pd.read_csv('ranking_por_estado.csv', names=['2018', '2019', '2020', 'Total'], header=0)

2018 2019 2020 Total
1    MG   MG   MG    MG
2    SC   SC   SC    SC
3    PR   PR   PR    PR
4    RJ   RJ   RJ    RJ
5    SP   RS   RS    RS
6    RS   SP   SP    SP
7    BA   BA   BA    BA
8    GO   GO   GO    GO
9    PE   ES   PE    ES
10   ES   PE   ES    PE

In [10]:
"""LETRA B"""

"""
Usei basicamente a mesma lógica da letra a, filtrei cada dataframe, um para cada ano, e depois fiz um para
o total. Nesse só foi preciso o value_counts para devolver a quantidade dos dias da semana. Então, concatenei todos
usando concat, como os indices eram iguais, representando os dias das semanas, coloquei axis=1 e fiz um único dataframe.

Na construção do arquivo csv, deu o mesmo que na letra a, apresentava o unamed na primeira coluna, então, quando fiz o read
tratei especificando os "names" além de colocar o header=0ler o arquivo csv que eu criei.

"""

todos_anos = pd.concat([ano_2018, ano_2019, ano_2020], ignore_index=True)

ano_2018_filtrado = pd.DataFrame(ano_2018['dia_semana'].value_counts())
ano_2019_filtrado = pd.DataFrame(ano_2019['dia_semana'].value_counts())
ano_2020_filtrado = pd.DataFrame(ano_2020['dia_semana'].value_counts())
anos_filtrados = pd.DataFrame(todos_anos['dia_semana'].value_counts())
                              
df_concatenado = pd.concat([ano_2018_filtrado, ano_2019_filtrado, ano_2020_filtrado, anos_filtrados], axis=1)
df_concatenado.columns = ['2018', '2019', '2020', 'Total']
df_concatenado.to_csv('acidentes_por_dia_semana.csv')

In [11]:
pd.read_csv('acidentes_por_dia_semana.csv', names=['2018', '2019', '2020', 'Total'], header=0)

2018   2019   2020  Total
domingo        11283  11298  10420  33001
sábado         11239  11099  10667  33005
sexta-feira    10632  10381   9764  30777
segunda-feira   9422   9134   8519  27075
quinta-feira    9230   8755   8244  26229
quarta-feira    8824   8374   8155  25353
terça-feira     8665   8405   7779  24849

### 4ª Tarefa ###

Com relação a ocorrência dos acidentes, responda:
    
    (a) qual a causa mais frequente e a mais rara de acidentes registrados nos três anos?
    (b) qual a proporção de pessoas ilesas e de feridos graves por mês em cada um dos anos considerados? (proporção = numero de ilesos ou feridos graves/ número total de pessoas envolvidas no acidente);
    (c) mostre, reorganizando o conjunto de dados, se a pandemia conseguiu diminuir ou não a incidência de acidentes nas rodovias federais.

In [12]:
"""LETRA A"""

"""
Para saber a causa mais frequente e a mais rara utilizei o value counts na coluna de "causa_acidente", após isso, dei um .index
para tranformar e array, dessa forma, selecionando o indice [0] e o [-1] do array, já temos o mais frequente e mais raro.
"""

todos_anos = pd.concat([ano_2018, ano_2019, ano_2020], ignore_index=True)

ano_2018_filtrado = pd.DataFrame(ano_2018['causa_acidente'].value_counts()).index
ano_2019_filtrado = pd.DataFrame(ano_2019['causa_acidente'].value_counts()).index
ano_2020_filtrado = pd.DataFrame(ano_2020['causa_acidente'].value_counts()).index
causa_acidente_filtr = pd.DataFrame(todos_anos['causa_acidente'].value_counts()).index


dicionario = {'2018': [ano_2018_filtrado[0], ano_2018_filtrado[-1]], '2019': [ano_2019_filtrado[0], ano_2019_filtrado[-1]],
       '2020': [ano_2020_filtrado[0], ano_2020_filtrado[-1]], 'Total': [causa_acidente_filtr[0], causa_acidente_filtr[-1]]}

DF = pd.DataFrame(dicionario, index=['Mais Frenquente', 'Mais raro'])
DF

2018  \
Mais Frenquente          Falta de Atenção à Condução   
Mais raro        Ingestão de Substâncias Psicoativas   

                                                2019  \
Mais Frenquente          Falta de Atenção à Condução   
Mais raro        Ingestão de Substâncias Psicoativas   

                                                   2020  \
Mais Frenquente             Falta de Atenção à Condução   
Mais raro        Pedestre cruzava a pista fora da faixa   

                                                          Total  
Mais Frenquente                     Falta de Atenção à Condução  
Mais raro        Acumulo de areia ou detritos sobre o pavimento

In [13]:
"""LETRA B"""

"""
PROFESSOR, ESSA QUESTÃO FOI LUTA! Sei que ela está longe de ser eficaz, mas foi a minha maneira de conseguir resolve-la,
usei muitos dicionários para separar cada categoria, por exemplo, para os ilesos em 2018, fiz isso para os três anos que iriamos
comparar -2018, 2019 e 2020-, para essas três categorias: ilesos, feridos graves e pessoas presentes no acidente. Dessa forma,
ficou mais fácil para mim resolver a questão.

Acho que a parte que usou mais do meu raciocínio foi justamente a de dividir os meses do ano, no primeiro for do código fiz
um range com o ano filtrado, que é justamente o concat de todos os anos com as colunas que vamos usar nessa questão
['data_inversa', 'feridos_graves', 'ilesos', 'pessoas'], então, nesse for utilizei o split da data_inversa a cada iteração,
guardando-a em uma variável de controle "x", entendi que com o split, o indice 1 da lista formada só poderia ser [01, 02, 03,
04, 05, 06, 07, 08, 09, 10, 11, 12], referentes aos meses do ano, então, fiz justamente os dicionarios tomando as keys como essas
posições, referenciando elas como meses do ano, então, usei 3 if's para classificar os anos, ou seja, o indice 0 da variável 'x',
que guarda o split, era referente ao ano -2018, 2019 ou 2020-, dessa forma, por conta do concat dos três anos, o ano da vez só
iria entrar no if específico, portanto, quando ele entrava em um desses if, começava a soma nas variáveis de controle dos
meses que criei nos dicionário separadamente, para fazer a soma na variável do mês específico, era só especificar a key
do split do 'x' índice 1, onde contém as variáveis dos meses iguais a da data_inversa referente aos meses.

Depois desse for, as próximas linhas foram para a criação dos três primeiros dataframes, onde contém os números literais
dos ilesos, feridos graves e pessoas envolvidas no acidente, separados em mês e ano.

O segundo for utilizei para criar a parte das proporções pedidas na questão, esse foi mais fácil, iterei o dicionário 
referente ao ano de 2018, mas só queria para tomar como base as chaves, que são iguais a todos os dicionários que eu criei
dessa forma, foi preciso somente fazer a divisão e colocar em uma formatação a dois decimos e colocar os valores das 
porcentagens nos novos dicionários que eu criei respectivos as proporções, por fim, criei o dataframe concatenando
as informações.

"""
from collections import Counter

dicionario2018_ilesas = {'01': 0, '02': 0, '03': 0, '04': 0, 
                  '05': 0, '06': 0, '07': 0, '08': 0, 
                  '09': 0, '10': 0, '11': 0, '12': 0}
dicionario2019_ilesas = dicionario2018_ilesas.copy()
dicionario2020_ilesas = dicionario2018_ilesas.copy()
dicionario2018_feridos = dicionario2018_ilesas.copy()
dicionario2019_feridos = dicionario2018_ilesas.copy()
dicionario2020_feridos = dicionario2018_ilesas.copy()
dicionario2018_pessoas = dicionario2018_ilesas.copy()
dicionario2019_pessoas = dicionario2018_ilesas.copy()
dicionario2020_pessoas = dicionario2018_ilesas.copy()

todos_anos = pd.concat([ano_2018, ano_2019, ano_2020], ignore_index=True)

ano_filtrado = todos_anos[['data_inversa', 'feridos_graves', 'ilesos', 'pessoas']]

for i in range(len(ano_filtrado['data_inversa'])):
    x = ano_filtrado['data_inversa'][i].split("-")
    if x[0] == '2018':
        dicionario2018_ilesas[x[1]] += ano_filtrado['ilesos'][i]
        dicionario2018_feridos[x[1]] += ano_filtrado['feridos_graves'][i]
        dicionario2018_pessoas[x[1]] += ano_filtrado['pessoas'][i]
    elif x[0] == '2019':
        dicionario2019_ilesas[x[1]] += ano_filtrado['ilesos'][i]
        dicionario2019_feridos[x[1]] += ano_filtrado['feridos_graves'][i]
        dicionario2019_pessoas[x[1]] += ano_filtrado['pessoas'][i]
    elif x[0] == '2020':
        dicionario2020_ilesas[x[1]] += ano_filtrado['ilesos'][i]
        dicionario2020_feridos[x[1]] += ano_filtrado['feridos_graves'][i]
        dicionario2020_pessoas[x[1]] += ano_filtrado['pessoas'][i]

DF2018_ilesas = pd.DataFrame(dicionario2018_ilesas, index=['2018'])
DF2019_ilesas = pd.DataFrame(dicionario2019_ilesas, index=['2019'])
DF2020_ilesas = pd.DataFrame(dicionario2020_ilesas, index=['2020'])

DF_total_ilesos = pd.concat([DF2018_ilesas, DF2019_ilesas, DF2020_ilesas])
DF_total_ilesos = DF_total_ilesos.rename(columns={'01':"Janeiro", '02': "Fevereiro", '03':"Março",'04':"Abril",'05':"Maio",'06':"Junho"
                         ,'07':"Julho",'08':"Agosto",'09':"Setembro",'10':"Outubro",'11':"Novembro",'12':"Dezembro"})

DF2018_feridos = pd.DataFrame(dicionario2018_feridos, index=['2018'])
DF2019_feridos = pd.DataFrame(dicionario2019_feridos, index=['2019'])
DF2020_feridos = pd.DataFrame(dicionario2020_feridos, index=['2020'])

DF_total_feridos = pd.concat([DF2018_feridos, DF2019_feridos, DF2020_feridos])
DF_total_feridos = DF_total_feridos.rename(columns={'01':"Janeiro", '02': "Fevereiro", '03':"Março",'04':"Abril",'05':"Maio",'06':"Junho"
                         ,'07':"Julho",'08':"Agosto",'09':"Setembro",'10':"Outubro",'11':"Novembro",'12':"Dezembro"})

DF2018_pessoas = pd.DataFrame(dicionario2018_pessoas, index=['2018'])
DF2019_pessoas = pd.DataFrame(dicionario2019_pessoas, index=['2019'])
DF2020_pessoas = pd.DataFrame(dicionario2020_pessoas, index=['2020'])

DF_total_pessoas = pd.concat([DF2018_pessoas, DF2019_pessoas, DF2020_pessoas])
DF_total_pessoas = DF_total_pessoas.rename(columns={'01':"Janeiro", '02': "Fevereiro", '03':"Março",'04':"Abril",'05':"Maio",'06':"Junho"
                         ,'07':"Julho",'08':"Agosto",'09':"Setembro",'10':"Outubro",'11':"Novembro",'12':"Dezembro"})

dicionario2018_propor_ilesas = dicionario2018_ilesas.copy()
dicionario2019_propor_ilesas = dicionario2018_ilesas.copy()
dicionario2020_propor_ilesas = dicionario2018_ilesas.copy()
dicionario2018_propor_feridos = dicionario2018_ilesas.copy()
dicionario2019_propor_feridos = dicionario2018_ilesas.copy()
dicionario2020_propor_feridos = dicionario2018_ilesas.copy()
for chave, valor in DF2018_ilesas.items():
    dicionario2018_propor_ilesas[chave] = "{:.2f}%".format((dicionario2018_ilesas[chave]/dicionario2020_pessoas[chave])*100)
    dicionario2019_propor_ilesas[chave] = "{:.2f}%".format((dicionario2019_ilesas[chave]/dicionario2020_pessoas[chave])*100)
    dicionario2020_propor_ilesas[chave] = "{:.2f}%".format((dicionario2020_ilesas[chave]/dicionario2020_pessoas[chave])*100)
    dicionario2018_propor_feridos[chave] = "{:.2f}%".format((dicionario2018_feridos[chave]/dicionario2020_pessoas[chave])*100)
    dicionario2019_propor_feridos[chave] = "{:.2f}%".format((dicionario2019_feridos[chave]/dicionario2020_pessoas[chave])*100)
    dicionario2020_propor_feridos[chave] = "{:.2f}%".format((dicionario2020_feridos[chave]/dicionario2020_pessoas[chave])*100)

DF2018_final_ilesos = pd.DataFrame(dicionario2018_propor_ilesas, index=['2018'])
DF2019_final_ilesos = pd.DataFrame(dicionario2019_propor_ilesas, index=['2019'])
DF2020_final_ilesos = pd.DataFrame(dicionario2020_propor_ilesas, index=['2020'])

DF_total_prop_ilesos = pd.concat([DF2018_final_ilesos, DF2019_final_ilesos, DF2020_final_ilesos])
DF_total_prop_ilesos = DF_total_prop_ilesos.rename(columns={'01':"Janeiro", '02': "Fevereiro", '03':"Março",'04':"Abril",'05':"Maio",'06':"Junho"
                         ,'07':"Julho",'08':"Agosto",'09':"Setembro",'10':"Outubro",'11':"Novembro",'12':"Dezembro"})

DF2018_final_feridos = pd.DataFrame(dicionario2018_propor_feridos, index=['2018'])
DF2019_final_feridos = pd.DataFrame(dicionario2019_propor_feridos, index=['2019'])
DF2020_final_feridos = pd.DataFrame(dicionario2020_propor_feridos, index=['2020'])

DF_total_prop_feridos = pd.concat([DF2018_final_feridos, DF2019_final_feridos, DF2020_final_feridos])
DF_total_prop_feridos = DF_total_prop_feridos.rename(columns={'01':"Janeiro", '02': "Fevereiro", '03':"Março",'04':"Abril",'05':"Maio",'06':"Junho"
                         ,'07':"Julho",'08':"Agosto",'09':"Setembro",'10':"Outubro",'11':"Novembro",'12':"Dezembro"})

print("Total de pessoas ilesas: ")
print("____________________________________________________________")
display(DF_total_ilesos)
print("Total de pessoas com ferimentos graves: ")
print("____________________________________________________________")
display(DF_total_feridos)
print("Total de pessoas presentes nos acidentes: ")
print("____________________________________________________________")
display(DF_total_pessoas)
print("Proporção de pessoas ilesas: ")
print("____________________________________________________________")
display(DF_total_prop_ilesos)
print("Proporção de pessoas com ferimentos graves: ")
print("____________________________________________________________")
display(DF_total_prop_feridos)

Total de pessoas ilesas: 
____________________________________________________________


Janeiro  Fevereiro  Março  Abril  Maio  Junho  Julho  Agosto  Setembro  \
2018     8554       6999   7276   5691  4851   5471   5958    5456      5480   
2019     5562       5004   5878   5327  5312   5683   6260    5544      5761   
2020     5847       5537   4321   3155  4043   4425   4934    5454      5486   

      Outubro  Novembro  Dezembro  
2018     5545      5719      6876  
2019     5836      5886      6581  
2020     5960      5491      6205

Total de pessoas com ferimentos graves: 
____________________________________________________________


Janeiro  Fevereiro  Março  Abril  Maio  Junho  Julho  Agosto  Setembro  \
2018     1510       1295   1539   1492  1315   1498   1502    1359      1504   
2019     1485       1260   1555   1358  1545   1631   1672    1566      1514   
2020     1512       1468   1213    989  1168   1212   1263    1535      1548   

      Outubro  Novembro  Dezembro  
2018     1452      1523      1717  
2019     1669      1535      1783  
2020     1736      1624      1836

Total de pessoas presentes nos acidentes: 
____________________________________________________________


Janeiro  Fevereiro  Março  Abril   Maio  Junho  Julho  Agosto  Setembro  \
2018    16665      14047  15110  13230  11489  12870  13618   12776     13027   
2019    13111      11799  13569  12815  12795  13411  14331   13421     13682   
2020    14070      13183  10775   8385  10074  10454  11628   12881     13187   

      Outubro  Novembro  Dezembro  
2018    12890     13197     15883  
2019    13931     13823     15585  
2020    14501     13490     15050

Proporção de pessoas ilesas: 
____________________________________________________________


Janeiro Fevereiro   Março   Abril    Maio   Junho   Julho  Agosto  \
2018  60.80%    53.09%  67.53%  67.87%  48.15%  52.33%  51.24%  42.36%   
2019  39.53%    37.96%  54.55%  63.53%  52.73%  54.36%  53.84%  43.04%   
2020  41.56%    42.00%  40.10%  37.63%  40.13%  42.33%  42.43%  42.34%   

     Setembro Outubro Novembro Dezembro  
2018   41.56%  38.24%   42.39%   45.69%  
2019   43.69%  40.25%   43.63%   43.73%  
2020   41.60%  41.10%   40.70%   41.23%

Proporção de pessoas com ferimentos graves: 
____________________________________________________________


Janeiro Fevereiro   Março   Abril    Maio   Junho   Julho  Agosto  \
2018  10.73%     9.82%  14.28%  17.79%  13.05%  14.33%  12.92%  10.55%   
2019  10.55%     9.56%  14.43%  16.20%  15.34%  15.60%  14.38%  12.16%   
2020  10.75%    11.14%  11.26%  11.79%  11.59%  11.59%  10.86%  11.92%   

     Setembro Outubro Novembro Dezembro  
2018   11.41%  10.01%   11.29%   11.41%  
2019   11.48%  11.51%   11.38%   11.85%  
2020   11.74%  11.97%   12.04%   12.20%

In [14]:
"""LETRA C"""

"""
Para criar os dataframes e separar os meses eu usei a mesma lógica da letra B, então, não irei me alongar muito nessa explicação
apenas uma pequena alteração foi feita, pois nessa só contei os números de acidentes por mês, dessa forma, só acrescentei
+= 1 a cada iteração do for. A discussão sobre a questão vai estar no tópico abaixo.
"""

todos_anos = pd.concat([ano_2019, ano_2020], ignore_index=True)

dicionario_acidentes_2019 = {'01': 0, '02': 0, '03': 0, '04': 0, 
                  '05': 0, '06': 0, '07': 0, '08': 0, 
                  '09': 0, '10': 0, '11': 0, '12': 0}
dicionario_acidentes_2020 = dicionario_acidentes_2019.copy()

for i in range(len(todos_anos)):
    x = todos_anos['data_inversa'][i].split("-")
    if x[0] == '2019':
        dicionario_acidentes_2019[x[1]] += 1
    elif x[0] == '2020':
        dicionario_acidentes_2020[x[1]] += 1


DF2019_acidentes = pd.DataFrame(dicionario_acidentes_2019, index=['2019'])
DF2020_acidentes = pd.DataFrame(dicionario_acidentes_2020, index=['2020'])

DF_total_acidentes = pd.concat([DF2019_acidentes, DF2020_acidentes])
DF_total_acidentes = DF_total_acidentes.rename(columns={'01':"Janeiro", '02': "Fevereiro", '03':"Março",'04':"Abril",'05':"Maio",'06':"Junho"
                         ,'07':"Julho",'08':"Agosto",'09':"Setembro",'10':"Outubro",'11':"Novembro",'12':"Dezembro"})
    
DF_total_acidentes

Janeiro  Fevereiro  Março  Abril  Maio  Junho  Julho  Agosto  Setembro  \
2019     5120       4963   5593   5436  5473   5578   5794    5703      5868   
2020     5497       5467   4790   3883  4657   4734   5127    5643      5689   

      Outubro  Novembro  Dezembro  
2019     5881      5802      6235  
2020     6120      5745      6196

Nessa questão separei os dados de acidentes em meses para ter uma base de analise melhor, mas já levando em conta o número total de acidentes no ano que eu retirei a partir da tarefa 2 letra a:

O números de acidente no ano de 2019 foi: 67446

O números de acidente no ano de 2020 foi: 63548

Já é possível ver uma diminuição do número de acidentes ocorridos, porém, ainda não é possível afirmar que a causa foi o covid.
De acordo com a OMS a pandemia do covid-19 começou a partir de 11 de março de 2020, então, analisando os meses de março até julho houve uma diminuição dos números de acidentes, após isso começou a se manter e aumentar como fevereiro. Com essa diminuição é possível INFERIR, mas não afirmar, que a pandemia do covid pode ter sim diminuido o número de acidentes. 